bigtable/google/cloud/bigtable/app_profile.py

```diff
@@ -0,0 +1,309 @@
+
+    @classmethod
+    def from_pb(cls, app_profile_pb, instance):
+        """Creates an instance app_profile from a protobuf.
+
+        :type app_profile_pb: :class:`instance_pb2.app_profile_pb`
+        :param app_profile_pb: An instance protobuf object.
+
+        :type instance: :class:`google.cloud.bigtable.instance.Instance`
+        :param instance: The instance that owns the cluster.
+
+        :rtype: :class:`AppProfile`
+        :returns: The AppProfile parsed from the protobuf response.
+        :raises: :class:`ValueError <exceptions.ValueError>` if the AppProfile
+                 name does not match
+                 ``projects/{project}/instances/{instance_id}/appProfiles/{app_profile_id}``
+                 or if the parsed instance ID does not match the istance ID
+                 on the client.
+                 or if the parsed project ID does not match the project ID
+                 on the client.
+        """
+        match_app_profile_name = (
+            _APP_PROFILE_NAME_RE.match(app_profile_pb.name))
+        if match_app_profile_name is None:
+            raise ValueError('AppProfile protobuf name was not in the '
+                             'expected format.', app_profile_pb.name)
+        if match_app_profile_name.group('instance') != instance.instance_id:
+            raise ValueError('Instance ID on app_profile does not match the '
+                             'instance ID on the client')
+        if match_app_profile_name.group('project') != instance._client.project:
+            raise ValueError('Project ID on app_profile does not match the '
+                             'project ID on the client')
+        app_profile_id = match_app_profile_name.group('app_profile_id')
+
+        result = cls(app_profile_id, instance)
+        result._update_from_pb(app_profile_pb)
+        return result
+
+    def _update_from_pb(self, app_profile_pb):
+        """Refresh self from the server-provided protobuf.
+        Helper for :meth:`from_pb` and :meth:`reload`.
+        """
+        self.routing_policy_type = None
+        self.allow_transactional_writes = None
+        self.cluster_id = None
+
+        self.description = app_profile_pb.description
+
+        routing_policy_type = None
+        if app_profile_pb.HasField('multi_cluster_routing_use_any'):
+            routing_policy_type = RoutingPolicyType.ANY
+            self.allow_transactional_writes = False
+        else:
+            routing_policy_type = RoutingPolicyType.SINGLE
+            self.cluster_id = app_profile_pb.single_cluster_routing.cluster_id
+            self.allow_transactional_writes = (
+                app_profile_pb.single_cluster_routing
+                .allow_transactional_writes)
+        self.routing_policy_type = routing_policy_type
+
+    def reload(self):
+        """Reload the metadata for this cluster"""
+
+        app_profile_pb = (
+            self._instance._client.instance_admin_client.get_app_profile(
```

Comment:
Please keep a reference to `self._instance._client.instance_admin_client` in self, and use that for all uses of `instance_admin_client`